aws-cpp-sdk-queues/source/sqs/SQSQueue.cpp

```diff
@@ -71,7 +71,8 @@ Message SQSQueue::Top() const
             if (receiveMessageOutcome.IsSuccess() && receiveMessageOutcome.GetResult().GetMessages().size() > 0)
             {
                 AWS_LOGSTREAM_DEBUG(CLASS_TAG, "Message found, returning");
-                return receiveMessageOutcome.GetResult().GetMessages()[0];
+                message = receiveMessageOutcome.GetResult().GetMessages()[0];
```

Comment:
I'd change `SetMaxNumberOfMessages(1)` to be configurable (add a ctor parameter with default = 1. And change the return type of `Top` to return a vector of messages. And in the loop check if the vector is empty to decide whether to sleep or not.